### PR TITLE
Imerica anchor to links

### PR DESCRIPTION
**Pull request by oorestisime**
#### What does this PR do?
Addresses #3297 

#### Where should the reviewer start?
Run tests. Observe they all pass. Observe that Anchor is still importable and backwards compatible.

#### What testing has been done on this PR?
Unit testing. All tests pass.

#### How should this be manually tested?
Import grommet into a new project and try using Link like you normally would Anchor. Also use Anchor like normal and anything that is upstream from Anchor.

#### Do the grommet docs need to be updated?
Theres definitely still vestiges of Anchor in the docs. I'll leave this PR as WIP for now.

#### Should this PR be mentioned in the release notes?
Yes.

#### Is this change backwards compatible or is it a breaking change?
Yes.

### CODE REVIEW

File: src/js/allcomponents.stories.js [hunk 1]
```diff
@@ -0,0 +1,353 @@
+import React, { Component } from 'react';
```
Comment (oorestisime) [inline src/js/allcomponents.stories.js]: hm this looks weird. we removed this file it seems from repo?

File: src/js/components/Link/StyledLink.js [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    `;
+  }
+  return css`
+    font-size: inherit;
+    line-height: inherit;
+  `;
+};
+
+// Deprecated: This will no longer be necessary on 3.0
+const getLinkTheme = theme => {
```
Comment (oorestisime) [inline src/js/components/Link/StyledLink.js]: this was my solution for backwards compatibility for theme

File: src/js/components/Link/__tests__/Link-test.js [hunk 3]
```diff
@@ -1,3 +1,4 @@
+/* eslint-disable jsx-a11y/anchor-is-valid */
```
Comment (oorestisime) [inline src/js/components/Link/__tests__/Link-test.js]: not sure why the moment i change Anchro to Link it understands its a link and expects better href.

**Comment by ShimiSun** [comment]
I'll be closing this PR, feel free to fix conflicts and reopen.

**Pull request closed without merging**